src/app/Shared/Services/Api.service.tsx

```diff
@@ -943,6 +1005,12 @@ export class ApiService {
     });
   }
 
+  downloadThreadDump(threadDump: ThreadDump): void {
+    this.ctx.url(threadDump.downloadUrl).subscribe((resourceUrl) => {
+      this.downloadFile(resourceUrl, threadDump.uuid);
```

Comment:
The second argument to `downloadFile` is supposed to help set a different filename for the downloaded file, which I think could be useful here.

```
this.downloadFile(resourceUrl, `${target.alias}_${threadDump.uuid}.thread_dump`);
```

or something might be a better filename for the user than the raw UUID.